python_modules/dagster/dagster_tests/asset_defs_tests/test_source_asset_observation_job.py

```diff
@@ -105,3 +108,97 @@ def bar(context):
             assets=[foo, bar],
             jobs=[define_asset_job("mixed_job", [foo, bar])],
         )
+
+
+@pytest.mark.parametrize(
+    "is_valid,resource_defs",
```

Comment:
The fact that most of the logic in the function depends on this boolean makes me think that it would be clearer to split it into two separate tests, and just define the asset at module level. Not critical feedback though.

Reply:
can't do that because of the closure over `executed`

Reply:
ah right - could have a shared function though. or it's a small enough amount of code it could just be copy/pasted. the `is_valid=False` one doesn't even need an implementation